js/main.js

```diff
@@ -1,45 +1,491 @@
+    for (let i = columns; i < columns + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[rows * LENGTH + i].classList.add(
+            "player-ship",
+            "horizontal"
+          )
+        : CPU_DIVS[rows * LENGTH + i].classList.add("cpu-ship", "horizontal");
+      board === PLAYER_BOARD
+        ? (PLAYER_DIVS[rows * LENGTH + i].id = `${ship.key}`)
+        : (CPU_DIVS[rows * LENGTH + i].id = `${ship.key}`);
+      ship.firstIndex = `${rows}${columns}`;
+      board[rows][i] = 1;
+    }
+  } else {
+    for (let i = rows; i < rows + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[i * LENGTH + columns].classList.add(
+            "player-ship",
+            "vertical"
+          )
+        : CPU_DIVS[i * LENGTH + columns].classList.add("cpu-ship", "vertical");
+      board === PLAYER_BOARD
+        ? (PLAYER_DIVS[i * LENGTH + columns].id = `${ship.key}`)
+        : (CPU_DIVS[i * LENGTH + columns].id = `${ship.key}`);
+      ship.firstIndex = `${rows}${columns}`;
+      board[i][columns] = 1;
+    }
+  }
+}
+
+//Handling the player clicking the CPU board to fire
+async function handleMove(evt) {
+  let clickedDiv = evt.target;
+  if (
+    shipCounter <= 6 ||
+    clickedDiv.classList.contains("miss") ||
+    clickedDiv.classList.contains("hit-ship") ||
+    clickedDiv.classList.contains("sunken-ship") ||
+    turn === -1 ||
+    winner !== null
+  )
+    return;
+
+  let divIdx = CPU_DIVS.indexOf(clickedDiv);
+  let rows = parseInt(divIdx.toString().charAt(0));
+  let columns = parseInt(divIdx.toString().charAt(1));
+  let currentShip;
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  if (CPU_BOARD[rows][columns] === 0) CPU_DIVS[divIdx].classList.add("miss");
+  if (clickedDiv.classList.contains("cpu-ship")) {
+    currentShip = findCorrectShip(CPU_BOARD, clickedDiv);
+    if (!clickedDiv.classList.contains("hit-ship")) currentShip.health--;
+    checkIfSunk(CPU_BOARD, currentShip, clickedDiv);
+    CPU_DIVS[divIdx].classList.add("hit-ship");
+    CPU_BOARD[rows][columns] = 2;
+  }
+  turn = -1;
+  await activeTurn();
+
+  cpuFire(PLAYER_BOARD);
+}
+
+//Generating the CPU firing locations
+function cpuFire(board) {
+  let randomDivIdx;
+
+  if (lastHitRow !== null && lastHitColumn !== null) {
+    suitableIndex = parseInt(getSuitableIndex(board));
+  }
+  if (
+    suitableIndex === null ||
+    isNaN(suitableIndex) ||
+    LOCATIONS.has(suitableIndex)
+  ) {
+    do {
+      randomDivIdx = getRandomNum(0, 99);
+    } while (LOCATIONS.has(randomDivIdx));
+  } else randomDivIdx = suitableIndex;
+
+  LOCATIONS.add(randomDivIdx);
+  let randomDiv = PLAYER_DIVS[randomDivIdx];
+  let rows = parseInt(randomDivIdx.toString().charAt(0));
+  let columns = parseInt(randomDivIdx.toString().charAt(1));
+
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  let currentShip;
+
+  if (PLAYER_DIVS[randomDivIdx].classList.contains("player-ship")) {
```

Comment:
Since we have the row/col indices here, why not check the board instead of the classes? Something like `if(PLAYER_BOARD[rows][columns] === 1)` then we know we're hitting a ship part that hasn't been hit already. This would also remove the need for the if statement on line 375. You could just do `currentShip.health--;` without any check.